README.md

```diff
@@ -43,118 +55,524 @@ Check the original technical report for an earlier version of the Model Extensio
+| data_type  | [Data Type Enum](#data-type-enum)      | **REQUIRED** The data type of values in the n-dimensional array. For model inputs, this should be the data type of the processed input supplied to the model inference function, not the data type of the source bands.   |
+
+A common use of `-1` for one dimension of `shape` is to indicate a variable batch-size.
+However, this value is not strictly reserved for the `b` dimension.
+For example, if the model is capable of automatically adjusting its input layer to adapt to the provided input data,
+then the corresponding dimensions that can be adapted can employ `-1` as well.
+
+#### Dimension Order
+
+Recommended values should use common names as much as possible to allow better interpretation by users and scripts
+that could need to resolve the dimension ordering for reshaping requirements according to the ML framework employed.
+
+Below are some notable common names recommended for use, but others can be employed as needed.
+
+- `batch`
+- `channel`
+- `time`
+- `height`
+- `width`
+- `depth`
+- `token`
+- `class`
+- `score`
+- `confidence`
+
+For example, a tensor of multiple RBG images represented as $B \times C \times H \times W$ should 
+indicate `dim_order = ["batch", "channel", "height", "width"]`.
+
+#### Normalize Enum
+
+Select one option from:
+- `min-max`
+- `z-score`
+- `l1`
+- `l2`
+- `l2sqr`
+- `hamming`
+- `hamming2`
+- `type-mask`
+- `relative`
+- `inf`
+- `clip`
+
+See [OpenCV - Interpolation Flags](https://docs.opencv.org/4.x/da/d54/group__imgproc__transform.html#ga5bb5a1fea74ea38e1a5445ca803ff121)
+for details about the relevant methods. Equivalent methods from other packages are applicable as well.
+
+When a normalization technique is specified, it is expected that the corresponding [Statistics](#bands-and-statistics)
+parameters necessary to perform it would be provided for the corresponding input.
+For example, the `min-max` normalization would require that at least the `minimum` and `maximum` statistic properties
+are provided, while the `z-score` would require `mean` and `stddev`.
+
+If none of the above values applies, `null` (literal, not string) can be used instead.
+If a custom normalization operation, or a combination of operations (with or without [Resize](#resize-enum)),
+must be defined instead, consider using a [Processing Expression](#processing-expression) reference.
+
+#### Resize Enum
+
+Select one option from:
+- `crop`
+- `pad`
+- `interpolation-nearest`
+- `interpolation-linear`
+- `interpolation-cubic`
+- `interpolation-area`
+- `interpolation-lanczos4`
+- `interpolation-max`
+- `wrap-fill-outliers`
+- `wrap-inverse-map`
+
+See [OpenCV - Normalization Flags](https://docs.opencv.org/4.x/d2/de8/group__core__array.html#ga87eef7ee3970f86906d69a92cbf064bd)
```

Comment:
I think this reference to normalization flags needs to be switched with the reference to interpolation.

long term I'd be interested in picking a different reference than OpenCV's C++ documentation, since the OpenCV lib is lower level than most folks encounter and the docs  are a bit hard to follow (python programmer might get confused with the C data types for example). But I think this is better than us rolling our own.